lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,419 @@
+            except CoordinateNotFoundError:
+                msg = ('new_cube does not have the coordinate in'
+                       ' the original cube {}'.format(coord_name))
+                raise CoordinateNotFoundError(msg)
+        return new_cube
+
+    @staticmethod
+    def calc_sin_phi(dtval, lats, lons):
+        """
+        Calculate sin of solar elevation
+
+        Args:
+            dtval (datetime.datetime):
+                Date and time.
+            lats (np.array):
+                Array 2d of latitudes for each point
+            lons (np.array):
+                Array 2d of longitudes for each point
+        Returns:
+            sin_phi (np.array):
+                Array of sine of solar elevation at each point
+
+        """
+        day_of_year = (dtval - datetime(dtval.year, 1, 1)).days
+        utc_hour = (dtval.hour * 60.0 + dtval.minute) / 60.0
+        sin_phi = calc_solar_elevation(lats, lons, day_of_year,
+                                       utc_hour, return_sine=True)
+        return sin_phi
+
+    @staticmethod
+    def calc_lats_lons(cube):
+        """
+        Calculate the lats and lons of each point for a given cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                cube containing x and y axis
+        Returns:
+            (tuple) : tuple containing:
+                **lats** (np.array):
+                    2d Array of latitudes for each point.
+                **lons** (np.array):
+                    2d Array of longitudes for each point.
+
+        """
+        trg_crs = lat_lon_determine(cube)
+        if trg_crs is not None:
+            xycube = next(cube.slices([cube.coord(axis='y'),
+                                       cube.coord(axis='x')]))
+            lats, lons = transform_grid_to_lat_lon(xycube)
+        else:
+            lats_row = cube.coord('latitude').points
+            lons_col = cube.coord('longitude').points
+            lats = np.repeat(lats_row[:, np.newaxis], len(lons_col), axis=1)
+            lons = np.repeat(lons_col[np.newaxis, :], len(lats_row), axis=0)
+        return lats, lons
+
+    def solar_interpolate(self, cube, interpolated_cube):
+        """
+        Interpolate solar radiation parameter which are zero if the
+        sun is below the horizon.
+
+        Args:
+            cube (iris.cube.Cube):
+                cube containing diagnostic cube valid at the beginning
+                of the period and cube valid at the end of the period
+            interpoldated_cube (iris.cube.Cube):
+                cube containing Linear interpolation of
+                cube at times in time_list.
+        Returns:
+            interpolated_cubes (iris.cube.CubeList):
+                A list of cubes interpolated to the desired times.
+
+        """
+
+        interpolated_cubes = iris.cube.CubeList()
+        (lats, lons) = self.calc_lats_lons(cube)
+        prev_data = cube[0].data
+        next_data = cube[1].data
```

Comment:
Does this need copying, given that you clear it later?

Reply:
This data is not cleared.  single_time.data is cleared.

Reply:
Thanks, this is made clearer when I understood that "cube" and "interpolated_cube" contain different data.  Could you make the docstring more explicit on this please, eg specify that interpolated cube does not contain the end points.

Reply:
The interpolated cube could contain the end points if that was in the time list. I've updated the docstring to try and make it more explicit that it as the interpolated times.